src/BlockInfo.cpp

```diff
@@ -802,6 +802,211 @@ void cBlockInfo::Initialize(cBlockInfoArray & a_Info)
+	a_Info[E_BLOCK_INACTIVE_COMPARATOR ].m_Hardness = 0.0f;
+	a_Info[E_BLOCK_ACTIVE_COMPARATOR   ].m_Hardness = 0.0f;
+	a_Info[E_BLOCK_DAYLIGHT_SENSOR     ].m_Hardness = 0.2f;
+	a_Info[E_BLOCK_BLOCK_OF_REDSTONE   ].m_Hardness = 5.0f;
+	a_Info[E_BLOCK_NETHER_QUARTZ_ORE   ].m_Hardness = 3.0f;
+	a_Info[E_BLOCK_HOPPER              ].m_Hardness = 3.0f;
+	a_Info[E_BLOCK_QUARTZ_BLOCK        ].m_Hardness = 0.8f;
+	a_Info[E_BLOCK_QUARTZ_STAIRS       ].m_Hardness = 0.8f;
+	a_Info[E_BLOCK_ACTIVATOR_RAIL      ].m_Hardness = 0.7f;
+	a_Info[E_BLOCK_DROPPER             ].m_Hardness = 3.5f;
+	a_Info[E_BLOCK_STAINED_CLAY        ].m_Hardness = 1.25f;
+	a_Info[E_BLOCK_STAINED_GLASS_PANE  ].m_Hardness = 0.3f;
+	a_Info[E_BLOCK_NEW_LEAVES          ].m_Hardness = 0.2f;
+	a_Info[E_BLOCK_NEW_LOG             ].m_Hardness = 2.0f;
+	a_Info[E_BLOCK_ACACIA_WOOD_STAIRS  ].m_Hardness = 2.0f;
+	a_Info[E_BLOCK_DARK_OAK_WOOD_STAIRS].m_Hardness = 2.0f;
+	a_Info[E_BLOCK_SLIME_BLOCK         ].m_Hardness = 0.0f;
+	a_Info[E_BLOCK_BARRIER             ].m_Hardness = -1.0f;
+	a_Info[E_BLOCK_IRON_TRAPDOOR       ].m_Hardness = 5.0f;
+	a_Info[E_BLOCK_PRISMARINE_BLOCK    ].m_Hardness = 1.5f;
+	a_Info[E_BLOCK_SEA_LANTERN         ].m_Hardness = 0.3f;
+	a_Info[E_BLOCK_HAY_BALE            ].m_Hardness = 0.5f;
+	a_Info[E_BLOCK_CARPET              ].m_Hardness = 0.1f;
+	a_Info[E_BLOCK_HARDENED_CLAY       ].m_Hardness = 1.25f;
+	a_Info[E_BLOCK_BLOCK_OF_COAL       ].m_Hardness = 5.0f;
+	a_Info[E_BLOCK_PACKED_ICE          ].m_Hardness = 0.5f;
+	a_Info[E_BLOCK_BIG_FLOWER          ].m_Hardness = 0.0f;
+	a_Info[E_BLOCK_STANDING_BANNER     ].m_Hardness = 1.0f;
+	a_Info[E_BLOCK_WALL_BANNER         ].m_Hardness = 1.0f;
+	a_Info[E_BLOCK_INVERTED_DAYLIGHT_SENSOR].m_Hardness = 0.2f;
+	a_Info[E_BLOCK_RED_SANDSTONE       ].m_Hardness = 0.8f;
+	a_Info[E_BLOCK_RED_SANDSTONE_STAIRS].m_Hardness = 0.8f;
+	a_Info[E_BLOCK_DOUBLE_RED_SANDSTONE_SLAB].m_Hardness = 2.0f;
+	a_Info[E_BLOCK_RED_SANDSTONE_SLAB  ].m_Hardness = 2.0f;
+	a_Info[E_BLOCK_SPRUCE_FENCE_GATE   ].m_Hardness = 2.0f;
+	a_Info[E_BLOCK_BIRCH_FENCE_GATE    ].m_Hardness = 2.0f;
+	a_Info[E_BLOCK_JUNGLE_FENCE_GATE   ].m_Hardness = 2.0f;
+	a_Info[E_BLOCK_DARK_OAK_FENCE_GATE ].m_Hardness = 2.0f;
+	a_Info[E_BLOCK_ACACIA_FENCE_GATE   ].m_Hardness = 2.0f;
+	a_Info[E_BLOCK_SPRUCE_FENCE        ].m_Hardness = 2.0f;
+	a_Info[E_BLOCK_BIRCH_FENCE         ].m_Hardness = 2.0f;
+	a_Info[E_BLOCK_JUNGLE_FENCE        ].m_Hardness = 2.0f;
+	a_Info[E_BLOCK_DARK_OAK_FENCE      ].m_Hardness = 2.0f;
+	a_Info[E_BLOCK_ACACIA_FENCE        ].m_Hardness = 2.0f;
+	a_Info[E_BLOCK_SPRUCE_DOOR         ].m_Hardness = 3.0f;
+	a_Info[E_BLOCK_BIRCH_DOOR          ].m_Hardness = 3.0f;
+	a_Info[E_BLOCK_JUNGLE_DOOR         ].m_Hardness = 3.0f;
+	a_Info[E_BLOCK_ACACIA_DOOR         ].m_Hardness = 3.0f;
+	a_Info[E_BLOCK_DARK_OAK_DOOR       ].m_Hardness = 3.0f;
+	// Synonym or ID compatibility
+	a_Info[E_BLOCK_YELLOW_FLOWER       ].m_Hardness = a_Info[E_BLOCK_DANDELION      ].m_Hardness;
```

Comment:
This shouldn't be necessary, since the value of the `E_BLOCK_YELLOW_FLOWER` constant is the same as `E_BLOCK_DANDELION` - this is effectively `a_Info[37].m_Hardness = a_Info[37].m_Hardness`.  (Same for the other synonyms)